### DISCUSSION TITLE
BT-1371 (Previous Planning Lot Identifier) does not seem "Pointless due to design"

### DISCUSSION THREAD

**Discussion by jpmckinney** [OP-TED/eForms-SDK#621] [Q&A] [answered]
Title quoted from https://docs.ted.europa.eu/eforms/1.0/schema/identifiers.html#pointlessDueToDesignSection

In the annex, under BG-125, there's:

* BT-125 Previous Planning Identifier (non-repeatable): "The identifier of a prior information notice or another similar notice related to this notice."
* BT-1251 Previous Planning Part Identifier (repeatable): "The identifier of a part of a prior information notice or another similar notice related to this notice."
* BT-1371 Previous Planning Lot Identifier (repeatable): "An identifier of one or more lots. In case of a prior information notice used only for information, the identifier of a part of the notice that may later become a lot or a self-standing procedure. The information in the previous planning section refers to this lot (or part) or these lots (or parts)."

eForms maps BT-125 to 2 fields, including BT-125(i)-Lot. However, there's no interpretation of BT-125 in which it refers to a lot. As I understand, eForms should instead rename BT-125(i)-Lot to (I assume) BT-1371-Lot.

Similarly, I'm not sure why the field BT-1251-Lot exists, since the business term BT-1251 is about a part, not a lot – but maybe there was some reason relating to PIN parts becoming CN lots.

**Comment by YvesJo** [comment, marked as answer]
Hi,
The Regulation Annex is a normalized representation of eForms. Level 1 Business Groups (*) related to lots are associated with lots using a dedicated Business Terms; (when information contained in the BG _ABC_ has to be associated with lots, BT _ABC Lot Identifier_ is defined for that purpose).
In the case of the BG _Previous Planning_, the BT _Previous Planning Lot Identifier_ is defined to associate an occurrence of one of its BG to the lot(s) it applies to.

The XML representation is, for its UBL part, a de-normalized one and for each lot, its related information is to be found in the lot itself.

- /*/cac:ProcurementProjectLot is the container for all the lot information
- **/*/cac:ProcurementProjectLot**/cac:TenderingProcess/cac:NoticeDocumentReference/cbc:ID contains BT-125 which is in the lot

The aim of BG-125 is to identify the PIN only notices (BT-125) and their parts (BT-1251) that combined together lead to the definition of a given lot (BT-1371). The lot (BT-1371) is to be identified in the current notice while the previous planning notices (BT-125) and their parts (BT-1251) are from the previous planning notices.

Please note that the trailing word in the field identifier is less about the object identified with the field than about the context the field appears in.
There are two fields associated to BT-125 (BT-125(i)-Lot & BT-125(i)-Part) as the Business Group may appear in different notices, including PIN only ones. PIN only notices contain Parts while other notices contain Lots, as a consequence the BG will appear in a Part or a Lot.
Similarly BT-1251-Lot will be used in any notice where BG-125 appears except for PIN only where BT-1251-Part is used.
KR

**Reply by jpmckinney** [reply]
Aha. In that case, I think what confused me is that BT-1371 is named "Previous Planning Lot Identifier", so I had interpreted this as meaning that the lot being identified was in a previous notice, not in the current notice as you describe.

Thank you for the explanation.
